crates/pathfinder/src/state/sync/class.rs

```diff
@@ -28,52 +27,65 @@ pub async fn download_class<SequencerClient: GatewayApi>(
+            Ok(DownloadedClass::Cairo { definition, hash })
+        }
+        starknet_gateway_types::class_hash::ComputedClassHash::Sierra(hash) => {
+            anyhow::ensure!(
+                class_hash == hash,
+                "Class hash mismatch, {} instead of {}",
+                hash,
+                class_hash.0
+            );
 
-                Ok(DownloadedClass::Cairo {
-                    definition,
-                    hash,
+            // FIXME(integration reset): work-around for integration containing Sierra classes
+            // that are incompatible with production compiler. This will get "fixed" in the future
+            // by resetting integration to remove these classes at which point we can revert this.
+            //
+            // The work-around ignores compilation errors on integration, and instead replaces the
+            // casm definition with empty bytes.
+            let (casm_definition, sierra_definition) =
+                tokio::task::spawn_blocking(move || -> (anyhow::Result<_>, _) {
+                    (
+                        crate::sierra::compile_to_casm(&definition, &version)
+                            .context("Compiling Sierra class"),
+                        definition,
+                    )
                 })
-            }
-            starknet_gateway_types::class_hash::ComputedClassHash::Sierra(hash) => {
-                anyhow::ensure!(
-                    class_hash == hash,
-                    "Class hash mismatch, {} instead of {}",
-                    hash,
-                    class_hash.0
-                );
+                .await?;
 
-                // FIXME(integration reset): work-around for integration containing Sierra classes
-                // that are incompatible with production compiler. This will get "fixed" in the future
-                // by resetting integration to remove these classes at which point we can revert this.
-                //
-                // The work-around ignores compilation errors on integration, and instead replaces the
-                // casm definition with empty bytes.
-                let casm_definition = crate::sierra::compile_to_casm(&definition, &version)
-                    .context("Compiling Sierra class");
-                let casm_definition = match (casm_definition, chain) {
-                    (Ok(casm_definition), _) => casm_definition,
-                    (Err(_), Chain::Integration) => {
-                        tracing::info!(class_hash=%hash, "Ignored CASM compilation failure integration network");
-                        Vec::new()
-                    }
-                    (Err(e), _) => return Err(e),
-                };
+            let casm_definition = match casm_definition {
+                Ok(casm_definition) => casm_definition,
+                Err(error) => {
+                    tracing::info!(class_hash=%hash, ?error, "CASM compilation failed, falling back to fetching from gateway");
```

Comment:
Should we make this a warning so we might get reports for it? I suppose we can rely on our tool instead. So this should be fine.